src/KnitPlugin.kt

```diff
@@ -103,6 +104,21 @@ open class KnitPluginExtension {
         ),
         files = files,
         rootDir = rootDir,
+        lineSeparator = evaluateLineSeparator(),
         check = check
     )
+
+    private fun evaluateLineSeparator(): String {
+        val ls = defaultLineSeparator
+        if (ls != null && ls != "\n" && ls != "\r\n" && ls != "\r") {
+            throw GradleException(
+                """Knit defaultLineSeparator must be one of:
+                |- Unix (\n)
+                |- Windows (\r\n)
+                |- Classic Mac OS (\r)
```

Comment:
This is rather my libraries-related ranting, so feel free to ignore or object :) 

'\r' seems to be a long-dead legacy (OS 9 and long dead classic OS X from 2002) and it's unlikely that someone either uses this machines or actually preserves their convention, so I always strongly object against adding `\r` to any new code

Reply:
I was unsure at first what to support, then I just followed what IntelliJ offers (bottom right corner in editor).

I'm fine with just supporting UNIX and WIN. However, one could argue now (even before, but more now) why to pass the line separator explicitly to the extension and not an enum. I think, usually it's awkward in a Gradle script to import enums. Another possibility now would be to switch the two cases with a boolean. Would you like that better?

Reply:
I think this is completely okay the way it is. Especially when we've changed the default and this option is just a _temporary_ migration mechanism